pkg/readiness/ready_tracker_unit_test.go

```diff
@@ -162,3 +214,60 @@ func Test_ReadyTracker_TryCancelTemplate_Retries(t *testing.T) {
 		t.Fatal("tracker with 0 retries and cancellation should be satisfied")
 	}
 }
+
+func Test_Tracker_TryCancelData(t *testing.T) {
+	l := lister{syncsets: true}
+	for _, tt := range []struct {
```

Comment:
Writing this is a range function leads to a bunch of unnecessary indentation and makes the test cases harder to differentiate from the test logic.

Please define the test cases and use `t.Run()`